src/main/webapp/app/exercises/shared/submission/submission.service.ts

```diff
@@ -46,6 +53,40 @@ export class SubmissionService {
         );
     }
 
+    /**
+     * Find the submissions with complaints for a tutor for a specified exercise (complaintType == 'COMPLAINT').
+     * @param exerciseId
+     */
+    getSubmissionsWithComplaintsForTutor(exerciseId: number): Observable<HttpResponse<SubmissionWithComplaintDTO[]>> {
+        return this.http
+            .get<SubmissionWithComplaintDTO[]>(`api/exercises/${exerciseId}/submissions-with-complaints`, { observe: 'response' })
+            .pipe(map((res) => this.convertDTOsFromServer(res)));
+    }
+
+    protected convertDTOsFromServer(res: HttpResponse<SubmissionWithComplaintDTO[]>) {
+        if (res.body) {
+            res.body.forEach((dto) => {
+                dto.submission = this.convertSubmissionDateFromServer(dto.submission);
+                dto.complaint = this.convertDateFromServerComplaint(dto.complaint);
+            });
+        }
+        return res;
+    }
```

Comment:
I think it would be better if this function signature is: `SubmissionWithComplaintDTO[] → SubmissionWithComplaintDTO[]` instead of `HttpResponse<SubmissionWithComplaintDTO[]> → HttpResponse<SubmissionWithComplaintDTO[]>`

Reply:
Technically you are correct, but to keep everything consistent with the other methods, I will keep the signature with the `HttpResponse`.